### DISCUSSION TITLE
SARIMAX model implementation in AG

### DISCUSSION THREAD

**Discussion by ZohrahV** [autogluon/autogluon#4323] [Q&A] [answered]
Hi, 

I would like to implement a SARIMAX model in autogluon so that I can train on the past covariates as well for a local model. Do I need a custom model or can use existing models already?

**Comment by shchur** [comment, marked as answer]
Hi @ZohrahV, you would need a custom model implementation for that since local models in AutoGluon currently don't support any covariates.

**Reply by ZohrahV** [reply]
Hi,

Thank you for the reply! Would you be able to guide me to the resources in order to implement a custom time series model?

**Reply by shchur** [reply]
This functionality is currently not very well documented, but here is a discussion thread that describes the main steps https://github.com/autogluon/autogluon/discussions/3091.

By the way, can you please share what SARIMAX implementation you would like to use that can handle past-only covariates?

**Reply by ZohrahV** [reply]
Thank you! I would like to implement this SARIMAX model from statsmodels: https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.sarimax.SARIMAX.html

**Reply by ZohrahV** [reply]
@shchur was there any particular reason why statistical models in AutoGluon currently don't support any covariates?

**Reply by shchur** [reply]
As far as I understand, SARIMAX from statsmodels only supports known covariates and not past covariates.

Adding covariates support for statistical models is on our roadmap, but we haven't gotten to it yet because some other items took higher priority.
